Adding Primitives for Music Notation Rendering via Redux

caiokf opened the pull request.

rplotkin commented on src/modules/tasks/reducers.js:
```diff
@@ -1,40 +1,33 @@
 // @flow
 
-import { List } from 'immutable'
+import { List } from 'immutable';
```
pretty sure lint is configured to have no semicolons. you should be able to run eslint --fix and have this back on track. If it's still complaining, I'll change a setting, so that semi's are not used.

rplotkin commented on src/views/music/components/Stave.js:
```diff
@@ -0,0 +1,41 @@
+  stave: Flow.Stave;
+  staveContainer: React.Element<any>;
+
+  static propTypes = {
```
props: Props instead, so we use Flow for checking instead of the runtime propTypes check.
Above the component, can define
```
type StateProps = {blah}
type DispatchProps = {blah}
type OwnProps = {blah}
type Props = StateProps & DispatchProps & OwnProps
```
(or something like that)

rplotkin commented on src/views/music/components/Stave.js:
```diff
@@ -0,0 +1,41 @@
+  }
+
+  componentDidMount() {
+    const width = this.props.width || 500
```
I prefer
`const { width = 500, height=500 } = this.props`

rplotkin commented on src/views/music/components/Stave.js:
```diff
@@ -0,0 +1,41 @@
+
+  render(): React.Element<any> {
+    return (
+      <div id={cuid()} className="stave" ref={(div) => { this.staveContainer = div }}>
```
do we need a cuid on each?

rplotkin replied:
can be a styled component

caiokf replied:
no. i was using it to assert against something on the element until i could test something else. since the elements are generated by the lib upon mounting, i had to change the enzyme method from `shallow` to `mount`, which required some jsdom setup. done now

caiokf replied:
maybe i need an example of how this would look like/what you are expecting

caiokf commented:
@rplotkin I am still addressing your comments in here, but inn the mean time, i also:
- created a git commit hook to run lint, flow and test before committing
- extracted eslint config to its own file

Please take a look and see if you like those changes
